src/main/java/uk/gov/hmcts/reform/userprofileapi/infrastructure/controllers/UserProfileController.java

```diff
@@ -141,64 +158,78 @@ public UserProfileController(UserProfileService<RequestData> userProfileService)
-            )
-        );
+        GetUserProfileWithRolesResponse response = userProfileService.retrieveWithRoles(new UserProfileIdentifier(EMAIL, email.toUpperCase()));
+        IdamRolesInfo idamRolesInfo = idamService.getUserById(response.getIdamId());
+        response.setRoles(idamRolesInfo.getRoles());
+
+        return ResponseEntity.ok(response);
     }
 
-    @ApiOperation("Retrieves user profile queried by idamId")
-    @ApiParam(name = "idamId", required = true)
+    @ApiOperation("Retrieves user profile queried by email or userId. If both provided email is preferred")
     @ApiResponses({
-        @ApiResponse(
-            code = 200,
-            message = "Representation of a user profile data",
-            response = String.class
-        ),
-        @ApiResponse(
-            code = 400,
-            message = "Bad Request",
-            response = String.class
-        ),
-        @ApiResponse(
-            code = 404,
-            message = "Not Found",
-            response = String.class
-        ),
-        @ApiResponse(
-            code = 500,
-            message = "Internal Server Error",
-            response = String.class
-        )
+            @ApiResponse(
+                    code = 200,
+                    message = "Representation of a user profile data",
+                    response = GetUserProfileResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "Bad Request"
+            ),
+            @ApiResponse(
+                    code = 404,
+                    message = "Not Found"
+            ),
+            @ApiResponse(
+                    code = 500,
+                    message = "Internal Server Error"
+            )
     })
-
     @GetMapping(
-        params = "idamId",
-        consumes = APPLICATION_JSON_UTF8_VALUE,
-        produces = APPLICATION_JSON_UTF8_VALUE
+            produces = APPLICATION_JSON_UTF8_VALUE
     )
     @ResponseBody
-    public ResponseEntity<UserProfileResource> getUserProfileByIdamId(@RequestParam String idamId) {
-        log.info("Getting user profile with idamId: {}", idamId);
-
-        requireNonNull(idamId, "idamId cannot be null");
-
-        return ResponseEntity.ok(
-            userProfileService.retrieve(
-                new UserProfileIdentifier(IDAMID, idamId)
-            )
-        );
+    public ResponseEntity<GetUserProfileResponse> getUserProfileByEmail(@ApiParam(name = "email", required = false) @RequestParam (value = "email", required = false) String email,
+                                                                     @ApiParam(name = "userId", required = false) @RequestParam (value = "userId", required = false) String userId) {
+
+        if (email == null && userId == null) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        if (email != null) {
+
+            log.info("Getting user profile with email: {}", email);
+
```

Comment:
move the validation to validator and please  avoid if multiple times here

Reply:
Done